### PR TITLE
fix: Address regression introduced in #21284

### LINKED ISSUE #21476: always return with this error. ["There was an error loading the tables"]
image
image

1. i follow the install instruction
2. i add a mysql database.
3. when i try to visit the tables. it bring this error. i try old version. still the same
4. what i miss in my install process?

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

Addresses regression introduced in https://github.com/apache/superset/pull/21284. 

cc: @EugeneTorap 

#### AFTER
Screen Shot 2022-09-14 at 11 22 08 AM
Screen Shot 2022-09-14 at 11 17 46 AM
Screen Shot 2022-09-14 at 11 16 17 AM

### TESTING INSTRUCTIONS

CI and verified the table selector worked in various surfaces.

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #21476
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/LeftPanel/index.tsx [hunk 1]
```diff
@@ -185,19 +185,6 @@ export default function LeftPanel({
     }
   }, [resetTables]);
 
-  const search = useMemo(
```
Comment (john-bodley) [inline superset-frontend/src/views/CRUD/data/dataset/AddDataset/LeftPanel/index.tsx:188]: I'm not entirely sure where this workflow is defined.

**Comment by john-bodley** [comment]
@eschutho, @lyndsiWilliams, and @pkdotson per https://github.com/apache/superset/pull/21075 it seems there's a new dataset selector which uses server side searching which was disabled in https://github.com/apache/superset/pull/21284. Is this actually a requirement for these new components or is it possible to remove said logic?

**Comment by lyndsiWilliams** [comment]
> @eschutho, @lyndsiWilliams, and @pkdotson per #21075 it seems there's a new dataset selector which uses server side searching which was disabled in #21284. Is this actually a requirement for these new components or is it possible to remove said logic?

There are actually pre-existing instances of paginated searching elsewhere in SQL Lab. Bringing this search functionality to the front end could get really expensive, could we re-enable server side searching?

**Comment by john-bodley** [comment]
@lyndsiWilliams per you comment, 

> There are actually pre-existing instances of paginated searching elsewhere in SQL Lab. Bringing this search functionality to the front end could get really expensive, could we re-enable server side searching?

could you elaborate where this is used? Also when we (Airbnb) tried using the backend search functionality—augmenting our local environment—for fetching tables names we noticed the latency was unacceptable. 

cc: @justin-park

**Comment by lyndsiWilliams** [comment]
> @lyndsiWilliams per you comment,
> 
> > There are actually pre-existing instances of paginated searching elsewhere in SQL Lab. Bringing this search functionality to the front end could get really expensive, could we re-enable server side searching?
> 
> could you elaborate where this is used? Also when we (Airbnb) tried using the backend search functionality—augmenting our local environment—for fetching tables names we noticed the latency was unacceptable.
> 
> cc: @justinpark

I looked into this further and discussed with Elizabeth, we actually don't use server side searching for this and it isn't used elsewhere in SQL Lab like I thought. The table data isn't stored in our database, we fetch it from their db and it's a "reflection" of their database, not actually scanning rows in a database.

So it should be fine to remove the server side searching after all 😁

**Review by eschutho — APPROVED** [review]

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/LeftPanel/index.tsx [hunk 2]
```diff
@@ -191,7 +191,7 @@ export default function LeftPanel({
         const encodeTableName =
           value === '' ? undefined : encodeURIComponent(value);
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/dataset/AddDataset/LeftPanel/index.tsx:192]:
```suggestion
        const encodeTableName =
          value === '' ? undefined : encodeURIComponent(value);
```

Comment (eschutho) [reply superset-frontend/src/views/CRUD/data/dataset/AddDataset/LeftPanel/index.tsx:192]: for test :)

**Comment by EugeneTorap** [comment]
@john-bodley any news?
FIY we have `/testenv up` command to deploy test ENV and then you can ping @jinghua-qa and @andrey-zayats to test it to avoid the regression bugs.

**Comment by EugeneTorap** [comment]
Fix #21476

**Comment by EugeneTorap** [comment]
@john-bodley Can add in description `Fix #21476` to auto close the issue?

**Review by lyndsiWilliams — APPROVED** [review]
LGTM! @jinghua-qa could you take a look at the ephemeral once it's up and make sure all tables are loading correctly now?

**Comment by eschutho** [comment]
I tested SQL lab in the ephemeral and tables are loading now.

**Comment by EugeneTorap** [comment]
Can we merge it?

**Comment by jinghua-qa** [comment]
LGTM!

**Comment by bumasl** [comment]
> /testenv up

alguien me puede explicar como puedo llevar a cabo en la ejecusion de este comando?

**Comment by eschutho** [comment]
@bumasl only Superset committers have permissions to create test environments. But you can request that someone start one for you if you'd like to test.

**Pull request merged**